core/src/wallet/bitcoin/synchronizers/BlockchainExplorerAccountSynchronizer.cpp

```diff
@@ -95,5 +135,191 @@ namespace ledger {
         std::shared_ptr<api::ExecutionContext> BlockchainExplorerAccountSynchronizer::getSynchronizerContext() {
             return getContext();
         }
+
+        int BlockchainExplorerAccountSynchronizer::putTransaction(soci::session &sql,
+                                                                  const BitcoinLikeBlockchainExplorerTransaction &transaction,
+                                                                  const std::shared_ptr<CommonBuddy> &buddy) {
+            // In case the transaction is in mempool, keep in memory before inserting it in database.
+            if (transaction.block.isEmpty()) {
+                std::static_pointer_cast<BitcoinSynchronizationBuddy>(buddy)->mempoolTransaction.emplace_back(
+                        transaction);
+                auto keychain = buddy->account->getKeychain();
+                int flag = 0;
+                // Mark inputs
+                for (auto& input : transaction.inputs) {
+                    if (input.address.hasValue()) {
+                        keychain->markAsUsed(input.address.getValue());
+                        flag |= BitcoinLikeAccount::FLAG_TRANSACTION_CREATED_SENDING_OPERATION;
+                    }
+                }
+                // Mark output
+                for (auto& output : transaction.outputs) {
+                    if (output.address.hasValue()) {
+                        keychain->markAsUsed(output.address.getValue());
+                        flag |= BitcoinLikeAccount::FLAG_TRANSACTION_CREATED_SENDING_OPERATION;
+                    }
+                }
+                return flag;
+            }
+            return buddy->account->putTransaction(sql, transaction);
+        }
+
+        std::shared_ptr<CommonBuddy>
+        BlockchainExplorerAccountSynchronizer::makeSynchronizationBuddy() {
+            return std::make_shared<BitcoinSynchronizationBuddy>();
+        }
+
+        inline bool isInputReplaced(const Input& input, uint32_t conflictingInputSequence) {
+            return input.sequence < conflictingInputSequence;
+        }
+
+        inline bool isInputReplaceable(const Input& input) {
+            return input.sequence < std::numeric_limits<uint32_t>::max();
+        }
+
+        template <template<class, class> class List, typename Alloc>
```

Comment:
Can we just have a normal std::vector in the parameters (and on all the places where we use std::list)?

Reply:
yes, I'll replace to std::vector